include/alp/omp/vector.hpp

```diff
@@ -0,0 +1,295 @@
+						}
+					}
+				}
+
+				/**
+				 * Copy constructor.
+				 *
+				 * @param other The vector to copy. The initialization state of the copy
+				 *              reflects the state of \a other.
+				 *
+				 * \parblock
+				 * \par Performance semantics.
+				 *      Allocates the same capacity as the \a other vector, even if the
+				 *      actual number of elements contained in \a other is less.
+				 *        -# This constructor entails \f$ \Theta(1) \f$ amount of work.
+				 *        -# This constructor allocates \f$ \Theta(\max{mn, cap} ) \f$ bytes
+				 *           of dynamic memory.
+				 *        -# This constructor incurs \f$ \Theta(mn) \f$ of data
+				 *           movement.
+				 *        -# This constructor \em may make system calls.
+				 * \endparblock
+				 *
+				 * \warning Avoid the use of this constructor within performance critical
+				 *          code sections.
+				 */
+				Vector( const Vector< T, omp > &other ) : Vector( other.n, other.cap ) {
+					initialized = other.initialized;
+					// const RC rc = set( *this, other ); // note: initialized will be set as part of this call
+					// if( rc != SUCCESS ) {
+					// 	throw std::runtime_error( "alp::Vector< T, omp > (copy constructor): error during call to alp::set (" + toString( rc ) + ")" );
+					// }
+				}
+
+				/**
+				 * Move constructor. The new vector equal the given
+				 * vector. Invalidates the use of the input vector.
+				 *
+				 * @param[in] other The GraphBLAS vector to move to this new instance.
+				 *
+				 * \parblock
+				 * \par Performance semantics.
+				 *        -# This constructor entails \f$ \Theta(1) \f$ amount of work.
+				 *        -# This constructor will not allocate any new dynamic memory.
+				 *        -# This constructor will use \f$ \Theta(1) \f$ extra bytes of
+				 *           memory beyond that at constructor entry.
+				 *        -# This constructor will move \f$ \Theta(1) \f$ bytes of data.
+				 * \endparblock
+				 */
+				Vector( Vector< T, omp > &&other ) : buffers( other.buffers ) {
+					other.buffers = nullptr;
+					// data_deleter = std::move( other.data_deleter );
+					// initialized = other.initialized; other.initialized = false;
+				}
+
+				/**
+				 * Vector destructor.
+				 *
+				 * \parblock
+				 * \par Performance semantics.
+				 *        -# This destructor entails \f$ \Theta(1) \f$ amount of work.
+				 *        -# This destructor will not perform any memory allocations.
+				 *        -# This destructor will use \f$ \mathcal{O}(1) \f$ extra bytes of
+				 *           memory beyond that at constructor entry.
+				 *        -# This destructor will move \f$ \Theta(1) \f$ bytes of data.
+				 *        -# This destructor makes system calls.
+				 * \endparblock
+				 *
+				 * \warning Avoid calling destructors from within performance critical
+				 *          code sections.
+				 */
+				~Vector() {
+					if( buffers != nullptr ) {
+						delete [] buffers;
```

Comment:
I think each buffer entry be deallocated (ie, need to  `delete[] buffers[ thread ]` ) before delete[]-ing buffers?